build/scripts/Benchmarking.fsx

```diff
@@ -10,26 +16,131 @@ open System.Linq
+        member val LowerFence=lowerFence with get, set
+        member val Q1=q1 with get, set
+        member val Median=median with get, set
+        member val Mean=mean with get, set
+        member val Q3=q3 with get, set
+        member val UpperFence=upperFence with get, set
+        member val Max=max with get, set
+        member val InterquartileRange=interquartileRange with get, set
+        member val Outliers=outliers with get, set
+        member val StandardError=standardError with get, set
+        member val Variance=variance with get, set
+        member val StandardDeviation=standardDeviation with get, set
+        member val Skewness=skewness with get, set
+        member val Kurtosis=kurtosis with get, set
+        member val ConfidenceInterval=confidenceInterval with get, set
+        member val Percentiles=percentiles with get, set
+
+    type Benchmark(displayInfo:string, namespyce:string, tipe:string, method:string, methodTitle:string, parameters:string,
+                   statistics:Statistics, memory:Memory) = 
+        member val DisplayInfo=displayInfo with get, set
+        member val Namespace=namespyce with get, set
+        member val Type=tipe with get, set
+        member val Method=method with get, set
+        member val MethodTitle=methodTitle with get, set
+        member val Parameters=parameters with get, set
+        member val Statistics=statistics with get, set
+        member val Memory=memory with get, set
+
+    type BenchmarkReport(title: string, totalTime:TimeSpan, date:DateTime, commit:string, host:HostEnvironmentInfo, benchmarks:Benchmark list) =
+        member val Title = title with get, set
+        member val TotalTime = totalTime with get, set
+        member val Date = date with get, set
+        member val Commit = commit with get, set
+        member val HostEnvironmentInfo = host with get, set
+        member val Benchmarks = benchmarks with get, set
+
+    let private testsProjectDirectory = Path.GetFullPath(Paths.Source("Tests"))
+    let private benchmarkOutput = Path.GetFullPath(Paths.Output("benchmarks")) |> directoryInfo
+    let private copyToOutput file = CopyFile benchmarkOutput.FullName file
+
+    let Run(runInteractive:bool) =
 
-   let Run() =
         ensureDirExists benchmarkOutput
+
         let projectJson = testsProjectDirectory @@ "project.json"
+
         // running benchmarks can timeout so clean up any generated benchmark files
         try
-            DotNetCli.Restore(fun p ->
-                { p with
-                    Project = projectJson
-                })
-
-            DotNetCli.RunCommand(fun p ->
-                { p with
-                    WorkingDir = testsProjectDirectory
-                }) "run Benchmark"
+            if runInteractive then
+                DotNetCli.RunCommand(fun p ->
+                    { p with
+                        WorkingDir = testsProjectDirectory
+                    }) "run -f net46 -c Release Benchmark"
```

Comment:
Should we hardcode `net46` here? The fake command line parser snoops `-all` off `target` names so we can e.g do `built test` and `build test-all` or `build integrate-all` which would tests all the TFM's we support.

Should we do something similar for `benchmark` ?

https://github.com/elastic/elasticsearch-net/blob/4fa7daef451f629286c4986672a6b1740a00eb55/build/scripts/Testing.fsx#L29